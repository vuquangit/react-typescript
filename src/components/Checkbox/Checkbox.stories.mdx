import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks'
import { action } from '@storybook/addon-actions'
import Checkbox from '.'
import Space from 'stories/components/styled-system/Space.mdx'
import Styles from 'stories/styles.mdx'

<Meta title="Story/Components/Checkbox" component={Checkbox} />

<Styles />

# Checkbox

With `MDX` we can define a story for `Checkbox` right in the middle of our
Markdown documentation.

export const Template = (args) => <Checkbox {...args} />

<Canvas>
  <Story
    name="Normal"
    args={{ variant: 'normal' }}
    onClick={action('Medium Checkbox clicked')}
  >
    {Template.bind({})}
  </Story>
</Canvas>

<div class="subheading">Props</div>

<Space />
