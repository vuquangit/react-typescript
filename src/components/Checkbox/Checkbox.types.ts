import { SpaceProps } from 'styled-system'

export type ICheckboxLabel = React.ButtonHTMLAttributes<HTMLLabelElement>

export interface ICheckboxProps {
  checked?: boolean
  label?: string
  color?: 'default' | 'primary' | 'secondary'
  size?: 'medium' | 'small'
  disabled?: boolean
  onChange?: () => void
}

export declare type BaseCheckboxProps = ICheckboxProps & SpaceProps
