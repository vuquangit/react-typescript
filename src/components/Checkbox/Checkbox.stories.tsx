import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import Wrapper from '@/test/supports/AppWrapper'

import Checkbox from '.'
import { ICheckboxProps } from './Checkbox.types'

export default {
  title: 'Story/Components/Checkbox',
  component: Checkbox,
  argTypes: {},
} as Meta

const Template: Story<ICheckboxProps> = (args) => (
  <Wrapper themeMode="light">
    <Checkbox {...args}>Checkbox story</Checkbox>
  </Wrapper>
)

export const Normal = Template.bind({})
Normal.args = {
  label: 'Label text ðŸŒ¼',
}
