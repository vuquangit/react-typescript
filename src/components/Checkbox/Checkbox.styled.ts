import styled from 'styled-components'
import { space } from 'styled-system'
import { ICheckboxLabel } from './Checkbox.types'

export const CheckboxWrapper = styled('div')`
  ${space};

  display: flex;
  justify-content: flex-start;
  align-items: center;
`

export const CheckboxInput = styled('input')`
  cursor: pointer;
  height: 16px;
  width: 16px;
`

export const CheckboxLabel = styled('label')<ICheckboxLabel>`
  padding-left: 8px;
  cursor: pointer;
`
