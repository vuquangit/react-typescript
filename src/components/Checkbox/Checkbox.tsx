import React, { FC } from 'react'

import {
  CheckboxInput,
  CheckboxLabel,
  CheckboxWrapper,
} from './Checkbox.styled'
import { BaseCheckboxProps } from './Checkbox.types'

const Checkbox: FC<BaseCheckboxProps> = ({
  checked,
  label,
  disabled,
  onChange,
}) => {
  return (
    <CheckboxWrapper data-testid="checkbox-wrapper">
      <CheckboxInput
        type="checkbox"
        id="checkboxId"
        checked={checked}
        onChange={onChange}
        disabled={disabled}
      />
      <CheckboxLabel htmlFor="checkboxId">{label}</CheckboxLabel>
    </CheckboxWrapper>
  )
}

export default Checkbox
