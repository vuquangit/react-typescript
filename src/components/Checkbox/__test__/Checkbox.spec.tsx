/**
 *
 * Tests for Checkbox
 *
 */

import React from 'react'
import { render /* fireEvent */ } from '@testing-library/react'

import Checkbox from '..'
import AppWrapper from '@/test/supports/AppWrapper'

describe('App Checkbox', () => {
  it('Test Checkbox with default', async () => {
    const { container, getByTestId } = render(
      <AppWrapper themeMode="light">
        <Checkbox />
      </AppWrapper>
    )

    expect(container).toMatchSnapshot()
    expect(getByTestId('checkbox-wrapper')).toBeInTheDocument()
  })
})
