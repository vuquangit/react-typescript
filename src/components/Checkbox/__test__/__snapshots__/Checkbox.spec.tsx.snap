// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Checkbox Test Checkbox with default 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  cursor: pointer;
  height: 16px;
  width: 16px;
}

.c2 {
  padding-left: 8px;
  cursor: pointer;
}

<div>
  <div
    class="c0"
    data-testid="checkbox-wrapper"
  >
    <input
      class="c1"
      id="checkboxId"
      type="checkbox"
    />
    <label
      class="c2"
      for="checkboxId"
    />
  </div>
</div>
`;
